File: Python/A_Project/A_Tic_Tac_Toe/GameStateHandler/ConcreteStates/o_won_state.py
from GameStateHandler.game_state import GameState


class OWonState(GameState):
    def next(self, context, player, has_won: bool):
        return

    def is_game_over(self) -> bool:
        return True

    def get_result_message(self) -> str:
        return "Player O has won the game!"

File: Ruby/A_Project/A_Tic_Tac_Toe/GameStateHandler/ConcreteStates/o_won_state.rb
require_relative '../game_state'

module GameStateHandler
  module ConcreteStates
    class OWonState
      include GameState

      def next(context, player, has_won)
        # Game is already over, no further state transitions
      end

      def game_over?
        true
      end

      def result_message
        "Player O has won the game!"
      end
    end
  end
end
